controllers/oauth2client_controller.go

```diff
@@ -205,16 +205,24 @@ func (r *OAuth2ClientReconciler) registerOAuth2Client(ctx context.Context, c *hy
 		return err
 	}
 
+	oauth2client, err := c.ToOAuth2ClientJSON()
+	if err != nil {
+		if updateErr := r.updateReconciliationStatusError(ctx, c, hydrav1alpha1.StatusRegistrationFailed, err); updateErr != nil {
+			return updateErr
+		}
+		return nil
```

Comment:
Shouldn't we return the original error here?

Reply:
fixed